internal/node/chainmanager.go

```diff
@@ -28,24 +28,28 @@ type chainManager struct {
 	tp      bus.TransactionPool
 	network *consensus.Network
 
-	close  chan struct{}
-	mu     sync.Mutex
-	tip    consensus.State
-	synced bool
+	close         chan struct{}
+	mu            sync.Mutex
+	lastBlockTime time.Time
```

Comment:
Keeping track of `lastBlockTime` prevents a deadlock when trying to fetch the current block from the consensus set while we're processing a consensus change. Consensus keeps a lock until all subscribers are notified, which was causing a deadlock. This'll soon be gone anyway so it doesn't really matter.